doc/rules/phpdoc/phpdoc_array_type.rst

```diff
@@ -10,7 +10,7 @@ Warning
 Using this rule is risky
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-Risky when ``array`` key should be present, but is missing.
```

Comment:
potential improvement as follow-up PR:

add test with union type and prevent type conversion if union type detected. (and drop risky flag)

Reply:
The problem is not union type in general, there is a test with union type.

The problem is that type `Foo|Bar[]` could mean `Foo|array<Bar>` (which is how it will be converted currently) or `Foo<Bar>` it `Foo` is `iterable` - exactly what happens [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.49.0/src/ConfigInterface.php#L95).

It was discussed in https://github.com/kubawerlos/php-cs-fixer-custom-fixers/issues/840

Reply:
Can the https://github.com/kubawerlos/php-cs-fixer-custom-fixers/issues/840 be added to the risky section in the docs directly? And possibly https://phpstan.org/try link. I consider myself as php developer with a lot of knowledge in this area but the current docs explanation is not clear enough to me to understand the risky part.

Reply:
further discussed in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7837#issuecomment-1943684588